src/materialized/tests/sql.rs

```diff
@@ -211,6 +211,35 @@ fn test_tail() -> Result<(), Box<dyn Error>> {
     Ok(())
 }
 
+#[test]
+fn test_tail_empty_upper_frontier() -> Result<(), Box<dyn Error>> {
+    ore::test::init_logging();
+
+    let config = util::Config::default();
+    let (_server, mut client) = util::start_server(config)?;
+
+    client.batch_execute("CREATE MATERIALIZED VIEW foo AS VALUES (1), (2), (3);")?;
+    // All records should be read into view before we start tailing.
+    thread::sleep(Duration::from_millis(100));
+
+    let mut tail_reader = client.copy_out("TAIL foo WITHOUT SNAPSHOT")?.split(b'\n');
+    let mut without_snapshot_count = 0;
+    while let Some(_value) = tail_reader.next().transpose()? {
+        without_snapshot_count += 1;
+    }
+    assert_eq!(0, without_snapshot_count);
+    drop(tail_reader);
```

Comment:
why do we need to call `drop` here explicitly? asking because I haven't seen that in our codebase before

Reply:
the `client.copy_out` is a mutable borrow of the client, so without the explicit `drop` of the `tail_reader` I am unable to reuse the same client on line 233!